File: arduinobot_ws/src/arduinobot_py_examples/arduinobot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
        def __init__(self):
                super().__init__("Simple_Publisher")
                self.pub_ =self.create_publisher(String,"Chatter",10)
                self.counter=0
                self.frequency=1.0
                self.get_logger().info("Publishing at %d Hz"% self.frequency)
                self.timer=self.create_timer(self.frequency,self.timerCallBack)

        def timerCallBack(self):
                msg=String()
                msg.data="Hello Ros2 -counter :%d" %self.counter
                self.pub_.publish(msg)
                self.counter +=1



def main():
        rclpy.init()
        simple_publisher=SimplePublisher()
        rclpy.spin(simple_publisher)
        simple_publisher.describe_node()
        rclpy.shutdown()

if __name__=="__main__":
        main()

File: arduinobot_ws/src/arduinobot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include  <std_msgs/msg/string.hpp>
#include <chrono>

using namespace std::chrono_literals;

class Simple_publisher :public rclcpp::Node
{
public:
    Simple_publisher() : Node("Simple_Publisher"), counter_(0)
    {
        create_publisher<std_msgs::msg::String>("Chatter",10);
        timer = create_wall_timer(ls,std::bind(&Simple_publisher::timerCallback,this));

        RCLCPP_INFO(get_logger(),"Publishing at 1Hz");
    }

    void timerCallback()
    {
        auto message=std_msgs::msg::String();
        message.data ="Helloo ROS 2 -couunter"+std::to_string(counter_++);
    }
private:
    unsigned int counter_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;

};




